lib/Frontend/ArgsToFrontendOutputsConverter.cpp

```diff
@@ -429,7 +429,7 @@ StringRef SupplementaryOutputPathsComputer::
   if (!outputFilename.empty() && outputFilename != "-")
     return outputFilename;
 
-  if (input.isPrimary() && input.file() != "-")
+  if (input.isPrimary() && input.file() != "-" && input.file() != "-e")
```

Comment:
Both lines 186 and 432 test if the file is "-" or is "-e". This should be factored. For that matter, I'm not comfortable with the use of '-e' for a filename, since it isn't a filename. (I know that it seems consistent with the use of '-', sigh.) At the very least, there needs to be a Doxygen comment where the `file` member function is defined and where the `InputFile.FileName` instance variable is defined. Better still would be a Kind enumeration for `InputFile`, sigh.

Reply:
We need to show *something* in diagnostics, but it wouldn't have to be "-e". We could do something like "<command-line>" instead, as long as it's very unlikely to be an actual filename. 

(And whatever it is, if it's referenced in more than one place I suppose it should probably be a string constant declared somewhere rather than a magic string. "-" is short enough and part of a cross-program idiom that we can probably get away with it, but "-e" is more specific.)

Reply:
Actually, since we're using `#sourceLocation` in some cases anyway, the name we use for that won't necessarily show up in diagnostics. We can still have diagnostics say "-e" without actually using that for the filename.